src/components/Trans.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+
+import { useI18n } from "../i18n";
+
+const REGEXP = /({{\w+?}})|(<\w+?>)|(<\/\w+?>)/g;
+const KEY_REGEXP = /{{(\w+?)}}/;
+const TAG_START_REGEXP = /<(\w+?)>/;
+const TAG_END_REGEXP = /<\/(\w+?)>/;
+
+const getTransChildren = (
+  format: string,
+  props: {
+    [key: string]: React.ReactNode | ((el: React.ReactNode) => React.ReactNode);
+  },
+) => {
+  const stack: { name: string; children: React.ReactNode[] }[] = [
+    {
+      name: "",
+      children: [],
+    },
+  ];
+
+  format.split(REGEXP).forEach((match) => {
+    if (match === undefined || match.length === 0) {
+      return;
+    }
+
+    const tagStartMatch = match.match(TAG_START_REGEXP);
+
+    if (tagStartMatch !== null) {
+      // Set the tag name as the name if it's one of the props, eg for "Please
+      // <link>click the button</link> to continue" tagStartMatch[1] = "link"
+      // and props contain "link" then it will be pushed to stack.
+      const name = tagStartMatch[1];
+      if (props.hasOwnProperty(name)) {
+        stack.push({
+          name,
+          children: [],
+        });
+      } else {
+        console.warn(
+          `Trans: missed to pass in variable start/end ${match} for interpolating ${format}`,
+        );
+      }
+
+      return;
```

Comment:
this shouldn't be needed if we use `else If` instead of multiple ifs and readability wise will be better as well

Reply:
It's a compromise between readability and performance. If you want to use `else if` you need to generate all regex matches before.

```js
const tagStartMatch = match.match(TAG_START_REGEXP);
const tagEndMatch = match.match(TAG_END_REGEXP);
const keyMatch = match.match(KEY_REGEXP);

if (tagStartMatch !== null) {

} else if (tagEndMatch != null) {

} else if (keyMatch !== null) {

} else {

}
```